test/core/pipeline/features/test_run.py

```diff
@@ -0,0 +1,50 @@
+from pytest_bdd import scenarios, given
+
+from haystack import Pipeline, component
+from haystack.core.errors import PipelineMaxLoops
+from haystack.testing.sample_components import AddFixedValue, Double
+from haystack.testing.factory import component_class
+
+
+scenarios("pipeline_run.feature")
+
+
+@given("a pipeline that has no components", target_fixture="pipeline_data")
+def pipeline_that_has_no_components():
+    pipeline = Pipeline()
+    inputs = {}
+    expected_outputs = {}
+    return pipeline, inputs, expected_outputs, []
```

Comment:
I'm thinking of using a dataclass to define the outputs of these functions.
It would make the code more readable and easier to understand for people that doen't have context on how these tests are ran.

Reply:
I gave it a try, more code for little to no gain.